samples/compiler_plugins/xnnpack/test/batch_matrix_multiply_e2e.mlir

```diff
@@ -0,0 +1,18 @@
+// RUN: iree-compile --iree-hal-target-backends=llvm-cpu --iree-plugin=xnnpack_sample %s | \
+// RUN: iree-run-module \
+// RUN:     --device=local-sync \
+// RUN:     --executable_plugin=$IREE_BINARY_DIR/samples/custom_dispatch/xnnpack/plugin/system_plugin$IREE_DYLIB_EXT \
+// RUN:     --module=- \
+// RUN:     --function=main \
+// RUN:     --input=2x2x2xf32=2 \
+// RUN:     --input=2x2x2xf32=4 | \
+// llvm-lit does not like checking for '[['
+// RUN:     sed 's/\[/(/g' | \
+// RUN: FileCheck %s --check-prefix=CHECK-SYSTEM
+
+// CHECK-SYSTEM: EXEC @main
+// CHECK-SYSTEM: 2x2x2xf32=((16 16](16 16]]((16 16](16 16]]
+func.func @main(%a : tensor<?x?x?xf32>, %b : tensor<?x?x?xf32>) -> tensor<?x?x?xf32> {
+  %c = xnnpack.batch_matrix_multiply %a, %b : (tensor<?x?x?xf32>, tensor<?x?x?xf32>) -> tensor<?x?x?xf32>
```

Comment:
We don't have to use a dynamic shape here. You are dealing with more difficult problems. :-P

We may need to discuss about the priority of the dynamic shape support (if it makes the problem much difficult.)

And I believe this is why you created `resultDims`. Am I right?

Reply:
The legalization pattern now works for static and dynamic shapes :) (this test now has dynamic and static sizes)